wordpress-mobile/WordPress-iOS
# Add a wrapper for Tenor API

This PR is the first part of solving the issue of adding Gif support using Google's Tenor API ( #12997 )

I've created a simple wrapper to replace the Giphy object in the Giphy's old SDK. Since the response we get from the Tenor's API has a completely different structure and there isn't 1-to-1 mapping between all the fields, I had to select suitable replacements among the fields that are available in the Tenor's API response. They might not be the best choices yet, that's why I haven't simplified the TenorMedia class further. I left some room for some experimentation  when the feature becomes testable in the UI of the app.

This PR doesn't have any user-visible changes

**To test:**

- Create an ApiKey by following the instructions in the documentation.
- Add the key into the credentials file .configure-files/wpcom_app_credentials

WPCOM_APP_ID=....
WPCOM_APP_SECRET=....
TENOR_APP_ID={The key you've just created}

- Rake dependencies
- Build and run, check it builds okay
